test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NetworkCallCountTest.cs

```diff
@@ -237,18 +245,29 @@ public void NetworkCallCount_RestoreLargePackagesConfigWithMultipleSourcesWithPa
                 };
 
                 // Act
-                var r = CommandRunner.Run(
-                    nugetexe,
-                    workingPath,
-                    string.Join(" ", args),
-                    waitForExit: true,
-                    timeOutInMilliseconds: (int)TimeSpan.FromMinutes(3).TotalMilliseconds);
+                CommandRunnerResult result;
+
+                try
+                {
+                    result = CommandRunner.Run(
+                        nugetexe,
+                        workingPath,
+                        string.Join(" ", args),
+                        waitForExit: true,
+                        timeOutInMilliseconds: (int)TimeSpan.FromMinutes(3).TotalMilliseconds);
+                }
+                catch (TimeoutException ex)
+                {
+                    LogDiagnosticData(ex);
+
+                    throw;
+                }
 
                 var packagesFolder = new LocalPackageRepository(packagesFolderPath);
                 var allPackages = packagesFolder.GetPackages().ToList();
 
                 // Assert
-                Assert.True(0 != r.Item1, r.Item2 + " " + r.Item3);
+                Assert.True(0 != result.Item1, result.Item2 + " " + result.Item3);
```

Comment:
I think this one is an actual restore problem with the test running under mono. My guess is that the machines that we have now are slower than the one we had earlier.

I don't see the reason for the 100 project count. The goal can be achieve with something like 75. 

https://github.com/NuGet/NuGet.Client/pull/2745/files#diff-39ee90a960e5a2713c35a976946ebf6eL135 
I change this in my draft PR for FrameworkReference but haven't pushed enough to see if it eliminated the flakiness